utils/healthcheck.py

```diff
@@ -105,20 +105,43 @@ def handle_fail(check):
 # Server checks
 @healthcheck(name="server working directory", remediation="", fatal=True)
 def check_curr_visit_dir() -> bool:
-    if "CURRENT_VISIT_DIR" not in os.environ:
+    # Check if current visit dir is valid 
+    visit_dir_env_var = "CURRENT_VISIT_DIR"
+    if visit_dir_env_var not in os.environ:
         check_curr_visit_dir.remediation = (
-            "CURRENT_VISIT_DIR environment variable not found"
+            f"{visit_dir_env_var} environment variable not found"
         )
         return False
-    if os.environ["CURRENT_VISIT_DIR"] == "":
-        check_curr_visit_dir.remediation = "CURRENT_VISIT_DIR is empty"
+    if os.environ[visit_dir_env_var] == "":
+        check_curr_visit_dir.remediation = f"{visit_dir_env_var} is empty"
         return False
-    current_visit_dir = Path(os.environ["CURRENT_VISIT_DIR"])
+    current_visit_dir = Path(os.environ[visit_dir_env_var])
     if not current_visit_dir.exists():
         check_curr_visit_dir.remediation = (
-            f"CURRENT_VISIT_DIR = {current_visit_dir} does not exist"
+            f"{visit_dir_env_var} = {current_visit_dir} does not exist"
```

Comment:
can you also include a check of whether the visit directory is writeable to the user who runs the server?
https://stackoverflow.com/questions/2113427/determining-whether-a-directory-is-writeable
Two good candidates for methods to do this are above: the access method, and "zak"'s method of writing a file and handling an exception if necessary

Reply:
Added in https://github.com/NSLS-II/lsdc/pull/299/commits/88b4ede2c957339cd59809ddb79b6a45199e2665